addons/sale_service/models/sale_service.py

```diff
@@ -1,63 +1,68 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from openerp.osv import fields, osv
-from openerp.tools.translate import _
+from openerp import api, fields, models, _
 
-class procurement_order(osv.osv):
+
+class ProcurementOrder(models.Model):
     _name = "procurement.order"
     _inherit = "procurement.order"
-    _columns = {
-        'task_id': fields.many2one('project.task', 'Task', copy=False),
-    }
 
-    def _is_procurement_task(self, cr, uid, procurement, context=None):
+    task_id = fields.Many2one('project.task', string='Task', copy=False)
+
+    @api.model
+    def _is_procurement_task(self, procurement):
```

Comment:
why pass procurement object in argument ? self itself a object of procurement...